Sprint 9. All Homework has been completed.

Maksim-Isaev opened the pull request:
Выполнено согласно ТЗ в теории. 
**_Для справки версия Android Studio Jellyfish | 2023.3.1 RC 2, built on April 5, 2024 _**

Maksim-Isaev committed:
Sprint 9. Homework 1.
- For Main, Media, Search, Settings activity created a styles.xml
- For Main, Media, Search, Settings activity fonts have been adjusted
- For Main activity has been checked the absence of shadows on the buttons

Maksim-Isaev committed:
Sprint 9. Homework 1.1
Optimised code before dark mode

Maksim-Isaev committed:
work in progress for dark mode

Maksim-Isaev committed:
Sprint 9. Homework 1 - completed

Maksim-Isaev committed:
Sprint 9. Homework 2 - completed
The buttons "Share the application", "Write to support", "User Agreement" are implemented

Maksim-Isaev committed:
Sprint 9. The switch color change is implemented

Maksim-Isaev committed:
Sprint 9. Homework 3 - The search bar is implemented.

Maksim-Isaev committed:
Add comments for SearchActivity.kt

Maksim-Isaev committed:
Sprint 9. Homework 3 is completed.
Updated the SearchActivity.kt.
SettingsActivity.kt some changes have been made

Maksim-Isaev committed:
Minor spelling corrections were made to the homework before submitting it after sprint 9.

AleksandrIlinskii commented:
Совет: в андроид разработке названия веток пишутся маленькими буквами

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -1,16 +1,109 @@
+    private var clearButtonVisibility = false
+    private var searchValue = TEXT_DEF
+
+    companion object {
```
По котлин код стайлу companion object пишут внизу файла 
https://kotlinlang.org/docs/coding-conventions.html#class-layout

Maksim-Isaev replied:
Выполнил корректировку, спасибо учту сохранил)

AleksandrIlinskii commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -1,16 +1,109 @@
+        setContentView(R.layout.activity_search)
+
+        // Восстанавливаем значение поиска, если оно было сохранено
+        if (savedInstanceState != null) {
```
Зачем это тут, если ты восстанавливаешь в onRestoreInstantState?

Maksim-Isaev replied:
Убрал лишний код

AleksandrIlinskii commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,100 @@
+<resources>
+    <!-- ________________________________Стили для Activity_main________________________________ -->
+
+    <style name="main_Layout_Button_Style" parent="Widget.MaterialComponents.Button.UnelevatedButton"> <!-- Стиль для кнопок главного экрана -->
```
Здесь и ниже
Обычно названия в xml пишут через все маленькие буквы
main_layout_button_style

AleksandrIlinskii replied:
Это относится не только к стилям, но и dimens, color, etc

Maksim-Isaev replied:
Скорректировал

Maksim-Isaev committed:
The position of the companion object in SearchActivity.kt has been adjusted according to the documentation

Maksim-Isaev committed:
The additional code has been removed.

Maksim-Isaev committed:
Corrected the writing of styles, themes, colors, strings.
The cancellation cross in the search field has been replaced.

Maksim-Isaev committed:
Adjustments have been made according to the comments:
1. The hint text is not in the strings.xml .
2. In the screen layout file, the EditText and the "Clear" button are not in the same FrameLayout (the button on top of the input field).
Minor additional edits have been made to the code.

Maksim-Isaev commented:
> Совет: в андроид разработке названия веток пишутся маленькими буквами

Спасибо учту. Я хотел в следующем спринте открыть новую ветку.
